### PR TITLE
Optimize BetterMending for less allocations by removing usage of streams

### PR DESCRIPTION
**Pull request by Vonr**
image
BetterMending was contributing significantly to allocations on my server even when idle (shown in picture) partially due to the usage of streams.
This PR seeks to resolve that by removing direct usages of streams in BetterMending.
There were also some other small changes to make it somewhat more efficient, such as directly checking if the entity is an instance of ItemEntity as well as short-circuiting the loop when the item has been fully repaired.

### CODE REVIEW

**Comment by Vonr** [comment]
Screenshot after this change for reference:
image

File: src/main/java/de/melanx/utilitix/content/BetterMending.java [hunk 1]
```diff
@@ -21,49 +18,52 @@
...
-                            int i = Math.min((int) (orb.getValue() * stack.getXpRepairRatio()), stack.getDamageValue());
-                            stack.setDamageValue(stack.getDamageValue() - i);
-                            orb.remove(Entity.RemovalReason.KILLED);
+        for (var entity : entities) {
+            if (!(entity instanceof ItemEntity item)) {
+                continue;
+            }
+
+            var stack = item.getItem();
+            if (stack.getDamageValue() == 0 || stack.getEnchantmentLevel(Enchantments.MENDING) == 0) {
```
Comment (MelanX) [inline src/main/java/de/melanx/utilitix/content/BetterMending.java]: Please change this to `<= 0`

File: src/main/java/de/melanx/utilitix/content/BetterMending.java [hunk 2]
```diff
@@ -21,49 +18,52 @@
...
-                .forEach(item -> {
-                    List<ExperienceOrb> xps = level.getEntitiesOfClass(ExperienceOrb.class, BoundingBoxUtils.expand(item, 7));
-                    for (ExperienceOrb orb : xps) {
-                        Vec3 vector = new Vec3(item.getX() - orb.getX(), item.getY() + (orb.getEyeHeight() / 2) - orb.getY(), item.getZ() - orb.getZ());
-                        if (vector.lengthSqr() < 0.2 && !level.isClientSide) {
-                            ItemStack stack = item.getItem();
-                            int i = Math.min((int) (orb.getValue() * stack.getXpRepairRatio()), stack.getDamageValue());
-                            stack.setDamageValue(stack.getDamageValue() - i);
-                            orb.remove(Entity.RemovalReason.KILLED);
+        for (var entity : entities) {
```
Comment (MelanX) [inline src/main/java/de/melanx/utilitix/content/BetterMending.java]: Please change this (and every following) `var` to the correct class (`Entity` in this case).

**Review by MelanX — CHANGES_REQUESTED** [review]
Looks good, thanks for the improvement. There are only two minor things I want you to address.

**Review by MelanX — APPROVED** [review]

**Pull request merged**